pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java

```diff
@@ -170,7 +174,12 @@ private void generateSwaggerForFinder(JSONObject pathObject, String routePath, F
 
         ArrayList<JSONObject> parameters = new ArrayList<JSONObject>();
 
-        for (Annotation[] annotations : method.getParameterAnnotations()) {
+        Annotation[][] parameterAnnotations = method.getParameterAnnotations();
+        Class<?>[] parameterTypes = method.getParameterTypes();
+        for (int i = 0; i < parameterTypes.length; i++) {
+          Class<?> parameterType = parameterTypes[i];
+          Annotation[] annotations = parameterAnnotations[i];
+
           if (annotations.length != 0) {
```

Comment:
This piece of code generates documentation based on the annotations (see the annotated Instance class). If there are no annotations on the parameter, then there isn't a place to pull the documentation from, and as such, we don't generate an empty documentation object in the documentation (which would probably not validate against the Swagger spec).